qiskit/utils/quantum_instance.py

```diff
@@ -967,12 +974,17 @@ def backend(self):
     @property
     def backend_name(self):
         """Return backend name."""
-        return self._backend.name()
+        if self._backend_interface_version <= 1:
+            return self._backend.name()
+        else:
+            return self._backend.name
 
     @property
     def is_statevector(self):
         """Return True if backend is a statevector-type simulator."""
-        return is_statevector_backend(self._backend)
+        if self._backend_interface_version <= 1:
+            return is_statevector_backend(self._backend)
+        return False
```

Comment:
Is there a reason here to test the backend version and return based on that? Since the method in backend utils handles both new and old, from what I can see, I would have thought no change was needed here. Bear in mind algorithms can/do adjust their behavior as to whether its a statevector or not - e.g adding measurements or not, and dealing with the result, so always returning False here for a newer backend, assuming they can still behave the same way, does not seem right.

Reply:
Good catch, this was leftover from an earlier debugging state while I was getting this working and I forgot to circle back after I updated `is_statevector_backend()` to support `BackendV2`. I've fixed this in: https://github.com/Qiskit/qiskit-terra/pull/7563/commits/1bca4dbced0d8ff2b71722a243c93d32b134d3e4